code/game/objects/items/tools/experimental_tools.dm

```diff
@@ -274,7 +276,7 @@
 		to_chat(user, SPAN_WARNING("You don't seem to know how to use \the [src]..."))
 		return
 
-	if(!pdcell || pdcell.charge == 0)
+	if(SEND_SIGNAL(src, COMSIG_CELL_CHECK_CHARGE) & COMPONENT_CELL_CHARGE_INSUFFICIENT)
```

Comment:
That's gonna sound dumb but if there's no cell at all then it works
Might want to reverse such signals so the default is having no cell if no component is attached

Reply:
The dialysis device will always have a cell component, and the proc inside the component for COMSIG_CELL_CHECK_CHARGE will return COMPONENT_CELL_CHARGE_INSUFFICIENT if there isn't a cell when one is needed.

Reply:
Yeah i meant the component, but it's true it's always present, makes sense 👍